packages/core-v2/src/visual-model/default-visual-model.ts

```diff
@@ -0,0 +1,183 @@
+import { LOCAL_VISUAL_MODEL } from "../model/known-models";
+
+import { MODEL_VISUAL_TYPE, ModelVisualInformation, VISUAL_GROUP_TYPE, VISUAL_NODE_TYPE, VISUAL_RELATIONSHIP_TYPE, VisualEntity, VisualGroup, VisualNode, VisualRelationship } from "./visual-entity";
+import {
+  WritableVisualModel,
+  OnPremiseUnderlyingVisualModel,
+  VisualModelListener,
+  HexColor,
+  RepresentedEntityIdentifier,
+  VisualModelData,
+} from "./visual-model";
+import { UnsubscribeCallback } from "./entity-model/observable-entity-model";
+import { EntityIdentifier } from "./entity-model/entity";
+
+/**
+ * This is how data were stored in the version one of the visualization model.
+ * Later we switch to using entities and underlying model.
+ */
+interface VisualModelJsonSerializationV1 {
+
+  type: typeof LOCAL_VISUAL_MODEL;
+
+  modelId: string;
+
+  visualEntities: Record<string, VisualEntity>;
+
+  modelColors: Record<string, string>;
+
+}
+
+export class DefaultVisualModel implements WritableVisualModel {
+
+  private model: OnPremiseUnderlyingVisualModel;
```

Comment:
Not sure if I understand what is represents. This (meaning the` model: OnPremiseUnderlyingVisualModel`) is the actual visual model which holds all the visual info and not the semantic one?

Reply:
The visual model is build on top of the entity model. The visual model provides definitions of visual entities and specific data processing. The underlying entity model provides operations on level of entities.

Basically the `model` holds entities and the `VisualModel` is capable of representing visual entities.